### PR TITLE
#322: Part 1, add tests for existing implementation on JavassistClassDeclaration

### PR DESCRIPTION
**Pull request by WimTibackx**
In preparation for the real fix for #322, add some test infrastructure to test symbolSolving on Javassist*Declarations, and add some tests for SymbolSolving on JavassistClassDeclaration.

### CODE REVIEW

File: java-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/SymbolSolverWithJavassistClassTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    public void testSolveSymbolInTypeCantResolveNonExistantField() {
+        SymbolReference<? extends ResolvedValueDeclaration> solvedSymbol = symbolSolver.solveSymbolInType(classDeclarationConcreteClass, "FIELD_THAT_DOES_NOT_EXIST");
+
+        assertFalse(solvedSymbol.isSolved());
+
+        try {
+            solvedSymbol.getCorrespondingDeclaration();
+        } catch (Exception e) {
+            assertTrue(e instanceof UnsupportedOperationException);
+            assertNull(e.getMessage());
```
Comment (ftomassetti) [inline java-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/SymbolSolverWithJavassistClassTest.java]: Not sure if the message should be null also in the future

File: java-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/SymbolSolverWithJavassistClassTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        fail("Expected UnsupportedOperationException when requesting CorrespondingDeclaration on unsolved SymbolRefernce");
+    }
+
+    @Test
+    public void testSolveSymbolInTypeCanResolveFieldInSuper() {
+        assertCanSolveSymbol("SUPER_FIELD", classDeclarationSubClassOwnJar);
+    }
+
+    @Test
+    @Ignore // TODO This fails at the moment due to issue #326
```
Comment (ftomassetti) [inline java-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/SymbolSolverWithJavassistClassTest.java]: Nice touch!

File: java-symbol-solver-testing/src/test/resources/javassist_symbols/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+For the tests regarding resolving symbols in Jar files, we need some jar files. Some other tests within JavaSymbolSolver use established external jars for that purpose. Given the very specific cases we need here, that would severly complicate writing and maintaining tests. Therefore, I've decided to write custom jars for the cases we need.
+
+`main_jar` contains most of the necessary classes, `included_jar` and `excluded_jar` both contain an interface and a superclass. Included_jar is included in the combined type solver, while excluded_jar is not.
```
Comment (ftomassetti) [inline java-symbol-solver-testing/src/test/resources/javassist_symbols/README.md]: Maybe in the future we can get Gradle tasks for that. For now this seems brilliant!

File: java-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/SymbolSolverWithJavassistClassTest.java (hunk 1 above)
Comment (WimTibackx) [reply java-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/SymbolSolverWithJavassistClassTest.java]: How does `CorrespondingDeclaration not available for unsolved symbol` sound?

Comment (ftomassetti) [reply java-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/SymbolSolverWithJavassistClassTest.java]: It sounds good

File: java-symbol-solver-testing/src/test/resources/javassist_symbols/README.md (hunk 3 above)
Comment (WimTibackx) [reply java-symbol-solver-testing/src/test/resources/javassist_symbols/README.md]: Currently looking into this, as I needed to add some more files and was tired of executing the commands manually. Seems like it should be more or less possible.

**Pull request merged**